### PR TITLE
Alter the views query so that we can get distinct result rows

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Alter the views query so that we can get distinct result rows

# Need Review By (Date)
-4/17 (if possible)

# Urgency
- medium

# Steps to Test
1. checkout this branch
1. clear cache
1. use devel generate to produce some publications
1. work with the view on `/admin/structure/views/view/hs_publications` to verify you get distinct results in each row even when there are multiple authors on some of the publications.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr563-hgwvtzmgst2dgved4c9cx9iqqq05gj0f.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e9634077fecf1f561d33579

Visual Diffs:
- https://dashboard.tugboat.qa/5e9634077fecf1f561d33579/visualdiffs

**Comment by pookmish** [comment]
@sherakama can i get a review on this ASAP

File: docroot/modules/humsci/hs_views_helper/hs_views_helper.module [hunk 1]
```diff
@@ -151,3 +153,62 @@ function hs_views_helper_preprocess_views_view_pattern(&$variables) {
...
+    // items to display. We will set those as the result rows to use.
+    $pagers = drupal_static('hs_views_helper_views_pre_execute');
+    if (!empty($pagers[$view->id()][$view->getDisplay()->display['id']])) {
+      $pager_settings = $pagers[$view->id()][$view->getDisplay()->display['id']];
+      $distinct_results = array_slice($distinct_results, $pager_settings['offset'], $pager_settings['limit']);
+    }
+
+    $i = 0;
+    // Reset the row indexes to produce accurate rendering.
+    foreach($distinct_results as &$row){
```
Comment (sherakama) [inline docroot/modules/humsci/hs_views_helper/hs_views_helper.module]:
```suggestion
    foreach($distinct_results as &$row) {
```

**Review by sherakama — APPROVED** [review]

File: docroot/modules/humsci/hs_views_helper/hs_views_helper.module (hunk 1 above)
Comment (pookmish) [reply docroot/modules/humsci/hs_views_helper/hs_views_helper.module]: Why didn't codeclimate pick that up? Maybe something sparkbox changed

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr563-hgwvtzmgst2dgved4c9cx9iqqq05gj0f.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e9634077fecf1f561d33579

Visual Diffs:
- https://dashboard.tugboat.qa/5e9634077fecf1f561d33579/visualdiffs

**Pull request merged**